src/main/java/org/onedatashare/server/service/UserService.java

```diff
@@ -45,13 +51,48 @@ public UserService(UserRepository userRepository) {
             .switchIfEmpty(Mono.error(new InvalidField("Invalid username or password")));
   }
 
-  public Object register(String email, String password, String passwordConfirm) {
-    User user = new User(email, password);
-    if(password.equals(passwordConfirm)) {
-      user.setValidationToken(user.validationToken());
-      return createUser(user).map(this::sendVerificationEmail);
-    }
-    else return Mono.error(new InvalidField("Passwords do not match")); //RuntimeException
+  public Object register(String email) {
+
+    return doesUserExists(email).flatMap(admin -> {
+
+      // Means admin user exists in the DB
+      if(admin.email!=null && admin.email.equals(email)) {
+        return Mono.just(new Response("User with email id already exists", 302));
+      }
+
+      String password = "e0bf258e-8814-46d5-875a-c060255f6664";
+    /*
+      This would be a same temporary password for each user while creating,
+      once the user goes through the whole User creation workflow, he/she can change the password.
+     */
+      User user = new User(email, password);
+      String token = user.validationToken();
+      user.setValidationToken(token);
+      String code = RandomStringUtils.randomAlphanumeric(6);
+      user.setVerifyCode(code);
+      user.registerMoment = new Date().getTime();
+      return createUser(user).flatMap(createdUser-> {
+        System.out.println("IN");
```

Comment:
Is this print statement needed? Can we take it off?

Reply:
Yes. Sorry, i am removing it